### DISCUSSION TITLE
Higher available clipboard size

### DISCUSSION THREAD

**Discussion by Azzurite** [LWJGL/lwjgl3#769] [Q&A] [answered]
So when you do `GLFW.glfwSetClipboardString` a new `MemoryStack` for that clipboard is created. That stack has the [default size of 64 kB](https://javadoc.lwjgl.org/org/lwjgl/system/Configuration.html#STACK_SIZE).

We want to be able to copy/paste larger content. Are there any problems to look out for when setting the `org.lwjgl.system.stackSize` higher? 

Since this is a general setting affecting all `MemoryStack`, not necessarily only applying to `glfwGet/SetClipboardString`, I don't know if this doesn't cause any negative effects if I set this to 512/1024/whatever.

**Comment by octylFractal** [comment, marked as answer]
For `glfwSetClipboardString` I would suggest allocating a separate byte buffer instead, and calling that overload instead of the `String` overload. For example:
```java
String forClipboard = "my very long string";
ByteBuffer buffer = MemoryUtil.memUTF8(forClipboard);
try {
  glfwSetClipboardString(window, buffer);
} finally {
  MemoryUtil.memFree(buffer);
}
// Note: I didn't test this code, there may be minor errors. The general idea should be sound.
```
Depending on how much you use it, it might be worth allocating a single buffer up-front and re-using it.

For `glfwGetClipboardString`, it doesn't use `MemoryStack` at all, it decodes the bytes directly into a `char[]` (or `byte[]` on Java 9+) which is fed into the `String` constructor. So you shouldn't see any limits from LWJGL with that.

**Reply by Azzurite** [reply]
Unfortunately we want to stay cross-platform and libGDX only uses the `CharSequence` method :/

**Reply by octylFractal** [reply]
You could just make the MemoryStack larger then, the only downside is that it will permanently allocate that much for every thread that touches the `MemoryStack` code, so e.g. 1024KB = 1MB = 1MB per thread that access `MemoryStack`, which can add up.

**Reply by Azzurite** [reply]
That was the question basically, what effect it would have. Thanks!

**Reply by SomeTroglodyte** [reply]
Increasing the System property just before a copy and tuning it back down immediately afterwards wouldn't work I assume, it will get read and stored long before?

**Reply by SomeTroglodyte** [reply]
Oh - "per thread that access MemoryStack," - rough estimate what proportion of candidates do?

**Reply by octylFractal** [reply]
> Increasing the System property just before a copy and tuning it back down immediately afterwards wouldn't work I assume, it will get read and stored long before?

No, it's set up on first use of a `MemoryStack` anywhere <https://github.com/LWJGL/lwjgl3/blob/master/modules/lwjgl/core/src/main/java/org/lwjgl/system/MemoryStack.java#L28>

> Oh - "per thread that access MemoryStack," - rough estimate what proportion of candidates do?

I'm not sure what you mean by that, if you mean what methods do, I don't think that's very useful information, but it's probably pretty low. Most LWJGL methods leave it to the user to allocate anything, and only the convenience methods that take e.g. `CharSequence` or return values from methods that normally take out buffers will be using it.

**Reply by SomeTroglodyte** [reply]
> not sure what you mean by that

Your answer fits nicely, thank you!
